client/rest-high-level/src/test/java/org/elasticsearch/client/FieldCapabilitiesIT.java

```diff
@@ -0,0 +1,133 @@
+            "        \"rating\": {" +
+            "          \"type\":  \"keyword\"" +
+            "        }" +
+            "      }" +
+            "    }" +
+            "  }" +
+            "}}",
+            ContentType.APPLICATION_JSON);
+        client().performRequest("PUT", "/" + "index2", Collections.emptyMap(), mappings);
+
+        StringEntity document2 = new StringEntity(
+            "{" +
+                "\"rating\": \"good\"," +
+                "\"title\": \"second title\"" +
+                "}",
+            ContentType.APPLICATION_JSON);
+        client().performRequest("PUT", "/index2/doc/1", Collections.emptyMap(), document2);
+
+        client().performRequest("POST", "/_refresh");
+    }
+
+    public void testBasicRequest() throws IOException {
+        FieldCapabilitiesRequest request = new FieldCapabilitiesRequest()
+            .indices("index1", "index2")
+            .fields("rating", "title");
+
+        FieldCapabilitiesResponse response = execute(request,
+            highLevelClient()::fieldCaps, highLevelClient()::fieldCapsAsync);
+
+        // Check the capabilities for the 'rating' field.
+        assertTrue(response.get().containsKey("rating"));
+        Map<String, FieldCapabilities> ratingResponse = response.getField("rating");
+        assertEquals(2, ratingResponse.size());
+
+        FieldCapabilities expectedKeywordCapabilities = new FieldCapabilities(
+            "rating", "keyword", true, true, new String[]{"index2"}, null, null);
+        assertEquals(expectedKeywordCapabilities, ratingResponse.get("keyword"));
+
+        FieldCapabilities expectedLongCapabilities = new FieldCapabilities(
+            "rating", "long", true, true, new String[]{"index1"}, null, null);
+        assertEquals(expectedLongCapabilities, ratingResponse.get("long"));
+
+        // Check the capabilities for the 'title' field.
+        assertTrue(response.get().containsKey("title"));
+        Map<String, FieldCapabilities> titleResponse = response.getField("title");
+        assertEquals(1, titleResponse.size());
+
+        FieldCapabilities expectedTextCapabilities = new FieldCapabilities(
+            "title", "text", true, false);
+        assertEquals(expectedTextCapabilities, titleResponse.get("text"));
+    }
+
+    public void testNonexistentFields() throws IOException {
+        FieldCapabilitiesRequest request = new FieldCapabilitiesRequest()
+            .indices("index2")
+            .fields("nonexistent");
+
+        FieldCapabilitiesResponse response = execute(request,
+            highLevelClient()::fieldCaps, highLevelClient()::fieldCapsAsync);
+        assertTrue(response.get().isEmpty());
+    }
+
+    public void testNoFields() {
+        FieldCapabilitiesRequest request = new FieldCapabilitiesRequest()
+            .indices("index2");
+
+        expectThrows(IllegalArgumentException.class,
+            () -> execute(request, highLevelClient()::fieldCaps, highLevelClient()::fieldCapsAsync));
+    }
```

Comment:
here we are testing request validation? shall we move this test to a unit test? for instance `FieldCapabilitiesRequestTests`?

Reply:
thanks